Doc/library/copyreg.rst

```diff
@@ -29,16 +29,13 @@ Such constructors may be factory functions or class instances.
 
    Declares that *function* should be used as a "reduction" function for objects
    of type *type*.  *function* should return either a string or a tuple
-   containing two or three elements.
+   containing two or three elements. See the :attr:`~pickle.Pickler.dispatch_table`
+   for more details on the interface of *function*.
 
-   The optional *constructor* parameter, if provided, is a callable object which
-   can be used to reconstruct the object when called with the tuple of arguments
-   returned by *function* at pickling time.  A :exc:`TypeError` is raised if the
-   *constructor* is not callable.
+   The optional *constructor* parameter, if provided, validates that
+   *constructor* is callable, and if uncallable then :exc:`TypeError` is raised.
```

Comment:
This raises the question: what is this argument good for? I would suggestt

```suggestion
   The *constructor* parameter is a legacy feature and it is now ignored.
```

Reply:
But since it can raise an exception, it's not exactly ignored. Maybe instead that '... and it should be ignored'?

Reply:
Maybe "The *constructor* parameter, if passed, must be a callable. It is a legacy feature and is now ignored."

Reply:
I do like immediately emphasizing that it's a legacy feature, so perhaps instead "The *constructor* parameter is a legacy feature and is now ignored, but if passed it must be a callable, otherwise TypeError is raised." The `TypeError` section is optional though as there's probably no need to go into too much detail.

Reply:
Sounds good! I think we can skip "otherwise TypeError is raised" as it's a fairly unimportant detail.